web/html/src/utils/functions.js

```diff
@@ -88,8 +88,37 @@ function sortByNumber(aRaw: Object, bRaw: Object, columnKey: string, sortDirecti
 }
 
 function sortByDate(aRaw: Object, bRaw: Object, columnKey: string, sortDirection: number): number {
-    const aDate = aRaw[columnKey] instanceof Date ? aRaw[columnKey] : new Date(aRaw[columnKey]);
-    const bDate = bRaw[columnKey] instanceof Date ? bRaw[columnKey] : new Date(bRaw[columnKey]);
+    /**
+     *  HACK
+     *
+     * Expected input String format: "YYYY-MM-DD HH:mm:ss z" where "z" is the timezone
+     * This is an 'unparsable' format for a javascript Date
+     *
+     * Assuming both dates are on the same timezone, we need to drop the timezone information in order to
+     * convert the String into a javascript Date and compare which one is before the other. In the end,
+     * if the initial assumption is true, the offset is the same.
+     * The troubles could start if the String format is not as expected or if the two timezones to compare are not the same.
```

Comment:
I think both of these cases can be detected and a suitable exception can be thrown. This can save us a lot of debugging time.

What you can do is:
 1. Make the timezone part an optional group in the regex, and see if both `a` and `b` match. If not, throw "invalid format"
 2. If either of them has a value for the timezone, check if they're equal. If not, throw "timezone mismatch"

Although it's possible to adjust timezones even if they don't match, that would be way out of scope, so let's just require the same timezones.

Oh, and you said the fix is **temporary**, so if this method's going to be deprecated soon anyway, please scratch all of the above :D